python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+            detector
+        yScale : `float`
+            Factor by which to normalize y-dimension. Corresponds to height of
+            detector
+
+        Returns
+        -------
+        outWCS : `lsst.afw.geom.SkyWcs`
+            WCS constructed from the input mappings
+        """
+        # Set up pixel frames
+        pixelFrame = astshim.Frame(2, "Domain=PIXELS")
+        normedPixelFrame = astshim.Frame(2, "Domain=NORMEDPIXELS")
+
+        if doNormalizePixels:
+            # Pixels will need to be rescaled before going into the mappings
+            normCoefficients = [-1.0, 2.0/xScale, 0,
+                                -1.0, 0, 2.0/yScale]
+            normMap = convertToAstPolyMapCoefficients(normCoefficients)
+        else:
+            normMap = astshim.UnitMap(2)
+
+        # All of the detectors for one visit map to the same tangent plane
+        tangentPoint = lsst.geom.SpherePoint(centerRA, centerDec)
+        cdMatrix = afwgeom.makeCdMatrix(1.0 * lsst.geom.degrees, 0 * lsst.geom.degrees, True)
+        iwcToSkyWcs = afwgeom.makeSkyWcs(lsst.geom.Point2D(0, 0), tangentPoint, cdMatrix)
+        iwcToSkyMap = iwcToSkyWcs.getFrameDict().getMapping("PIXELS", "SKY")
+        skyFrame = iwcToSkyWcs.getFrameDict().getFrame("SKY")
+
+        frameDict = astshim.FrameDict(pixelFrame)
+        frameDict.addFrame("PIXELS", normMap, normedPixelFrame)
+
+        currentFrameName = "NORMEDPIXELS"
+
+        # Assume python 3.7+ so dict is ordered:
+        for m, mapElement in enumerate(mapDict.values()):
+            mapType = mapElement['Type']
+
+            if mapType == 'Poly':
+                mapCoefficients = mapElement['Coefficients']
+                astMap = convertToAstPolyMapCoefficients(mapCoefficients)
+            elif mapType == 'Identity':
+                astMap = astshim.UnitMap(2)
+            else:
+                raise ValueError(f"Converting map type {mapType} to WCS is not supported")
+
+            if m == len(mapDict) - 1:
+                newFrameName = "IWC"
+            else:
+                newFrameName = "INTERMEDIATE" + str(m)
+            newFrame = astshim.Frame(2, f"Domain={newFrameName}")
+            frameDict.addFrame(currentFrameName, astMap, newFrame)
+            currentFrameName = newFrameName
+        frameDict.addFrame("IWC", iwcToSkyMap, skyFrame)
+
+        outWCS = afwgeom.SkyWcs(frameDict)
+        return outWCS
+
+    def _makeOutputs(self, wcsf, inputVisitSummary):
+        """Make a WCS object out of the WCS models.
+
+        Parameters
+        ----------
+        wcsf : `wcsfit.WCSFit`
+            WCSFit object, assumed to have fit model
+        inputVisitSummary : `lsst.afw.table.ExposureCatalog`
+            Catalogs with per-detector summary information from which to grab
+            detector information
+
+        Returns
+        -------
+        catalogs : `dict`
+            Dictionary of `lsst.afw.table.ExposureCatalog` objects with the WCS
```

Comment:
What are the keys and values of this dict?

https://developer.lsst.io/python/numpydoc.html#dictionary-types